### ISSUE TITLE
type index uses the wrong namespace

### ISSUE THREAD

**Issue by melvincarvalho** [solid/node-solid-server#1002]
As reported here https://github.com/solid/node-solid-server/issues/995#issuecomment-443675337

```
@prefix solid: <https://www.w3.org/ns/solid/terms#>.
<> a solid:TypeIndex
```

Is being produced in some type indexes.

My own is fine : https://melvin.solid.community/settings/publicTypeIndex.ttl

```
@prefix solid: <http://www.w3.org/ns/solid/terms#>.
```

But I can confirm that more recent accounts have the incorrect https namespace.  This is a **major** bug, and we need to urgently find the source.

**Comment by melvincarvalho** [comment]
Unfortunately this also means that we are possibly going to have to go into the server and change users' files in order to make the system work.  

This is the right thing to do on solid.community, imho.  Inrupt is going to have to decide whether they wish to make this decision.  I would recommend yes!  There may be some workarounds, but I cant think of one at the moment.

cc : @timbl @johnbizguy 

Hopefully not too many accounts are affected.  First step tho is to work out what is causing this, and how much of an issue it is.

**Comment by csarven** [comment]
I'd agree that it is quite likely a typo. Perhaps @dmitrizagidulin can quickly confirm that - perhaps more importantly, if there was a reason to be different than the http use of the ns.

https is also used in privateTypeIndex.ttl

re:
```
git log --follow -p default-templates/new-account/settings/publicTypeIndex.ttl
git log --follow -p default-templates/new-account/settings/privateTypeIndex.ttl
```

they were always using https. So, @melvincarvalho , if your publicTypeIndex.ttl used http, it is unlikely that it came from the templates. A simple explanation may be that you've changed it manually at some point?

**Comment by melvincarvalho** [comment]
Looks like the files are here : 

https://github.com/solid/node-solid-server/blob/master/default-templates/new-account/settings/privateTypeIndex.ttl

https://github.com/solid/node-solid-server/blob/master/default-templates/new-account/settings/publicTypeIndex.ttl

**Comment by melvincarvalho** [comment]
> A simple explanation may be that you've changed it manually at some point?

Could well be!  I can do some further analysis to confirm.  That might actually make life a bit easier.

Do note : the canonical prefixes are clearly articulated in the vocab README

https://github.com/solid/vocab#created-for-solid

**Comment by melvincarvalho** [comment]
So the steps needed : 

- [ ] Fix the templates
- [ ] Fix the tests
- [ ] Priority push to master
- [ ] Update servers code
- [ ] Update server user accounts, as appropriate

Anything else?

I'd like to get this fixed on solid.community asap or else apps will not work or work inconsistently.

**Comment by melvincarvalho** [comment]
As a stop gap I think I can manually edit the template files to fix the typo, and write a script to substitute the the right namespace in all ~/settings/ directories that I find.

Thoughts welcome!

**Comment by csarven** [comment]
Acknowledged. I don't doubt the http use at all - as it is by the far the majority of the uses in code/data out there. As I've mentioned elsewhere, it was always http to the best of knowledge. I recall discussing at some point (somewhere) but we decided to stick to http. The README, the base in the ns, type registries... came much later.

**Comment by melvincarvalho** [comment]
Quick 5 minute script to fix this issue :  

https://gist.github.com/melvincarvalho/b6c7121f2c239c1878a26e9465fa1166

Im sure it can be improved upon.

**Comment by melvincarvalho** [comment]
I am now able to fix solid.community but have not done so yet, I'll coordinate this with @ericprud who I've run the plan by.

Only concern raised is that we dont want to be out of sync with whatever is on github.

@kjetilk is there a way we can get a patch into the master of 4.4.0?  -- we're currently running 4.2.0 which is a side issue, but would like to upgrade.

**Comment by melvincarvalho** [comment]
So, I think this is quite a tricky one to fix.  @ericprud has sensibly suggested to have any changes in github.  What I'd also like to do is keep an audit trail of the steps.  So I've done a few things.

- [x] I've started an org for solid.community.  Anyone that would like to help out with the pod please ask [here](https://github.com/solid-community/node-solid-server/issues/1) to be added to the org.  

- [ ] This repo will contain any changes we make.  In this case we will add a patch to the templates so that solid terms goes from https -> http for new users.

- [x] I've created a very rough scripts repo that will contain scripts that perform fixes and help maintain a pod.  The first one is [fixsolidterms.sh](https://github.com/solid-community/scripts/blob/master/fixsolidterms.sh).  It's very raw at the moment, but I've tried it on my personal pod and it works.

- [ ] What I'm currently worried about is whether or not special characters in user names could be an issue.  I'll do some more testing before running it.

- [x] I've created a [wiki](https://github.com/solid-community/node-solid-server/wiki) that we can hopefully update with useful information about the pod.

- [ ] One thing I would like to do is that every time we are forced to alter users' files, we can add an entry to the wiki saying what we did, and why we did it.  I think audit and transparency are important to be aligned with solid's principles.

- [ ] Ideally, would like to align this with inrupt, because at the moment type indexes and apps that use them are likely to be broken.  So it's going to be advantageous if we get a consistent app experience across pods, wrt finding data.  Fortunately not too many apps (that I know of) do this right now, but in future they are likely to.  

I hope this addresses as many concerns on this issue as I can, feedback welcome!

**Comment by kjetilk** [comment]
@melvincarvalho Yes, I think it this is something we can do a release off of master of, e.g. 4.4.1 once we have a PR for it. 

Meeting week now, but I'll try to catch up on the bugs when I can.

**Comment by kjetilk** [comment]
> I'll put in a PR for the change. 

Any chance you could do submit it soon, @melvincarvalho ? I'd like to get it into 5.0.0 too as soon as possible.

**Comment by melvincarvalho** [comment]
@kjetilk should be addressed in https://github.com/solid/node-solid-server/pull/1027

LGTM

**Comment by kjetilk** [comment]
I'll close this now, since it is fixed in the 5.0.0 beta, which is what I'm pursuing. If someone wants to backport it to 4.x, please do so, we can certainly make a release, but there is little focus on that now.

**Comment by melvincarvalho** [comment]
Thanks again for working on this PR.

I believe, there is a consensus to move solid.community to 4.4, in order to benefit from the new quota system, hopefully I can free up some time to help with that.

@kjetilk I thought the plan was to add this the 4.4 branch?  Fixing type indexes is beneficial in order for apps to do the right thing.  

> If someone wants to backport it to 4.x, please do so, we can certainly make a release, but there is little focus on that now

OK, I'll give it a try!  If you could let me know the release process, Ill see if im able to create a 4.4.1 release, then.  Or just walk me through the steps and I'll go through them, as best I can.

Do you think it would also be nice for the community pod to coordinate with inrupt.net so that there is a consistent user and developer experience?  I think this has generally been considered desirable.  If so, do you have any info on what version inrupt.net is running (or how I can find out -- ie 4.3 or 4.4) and upgrade plans.  Thanks in advance!
